fog/sql_recovery_db/src/lib.rs

```diff
@@ -323,27 +323,33 @@ impl RecoveryDb for SqlRecoveryDb {
         let conn = self.pool.get()?;
 
         use schema::ingress_keys::dsl;
+        let last_scanned_block = diesel::dsl::sql::<diesel::sql_types::BigInt>(
+                    "(SELECT MAX(block_number) FROM ingested_blocks WHERE ingress_keys.ingress_public_key = ingested_blocks.ingress_public_key)"
```

Comment:
Instead of having to worry about the NULL scenario, I think you could use [COALESCE](https://www.postgresqltutorial.com/postgresql-coalesce/) e.g. `SELECT COALSECE(MAX(block_umber), 0)...`.
Not sure if that makes things simpler or not but it will remove the need for the `is_not_null()` filter down below.

Reply:
I like the usage of COALESCE to simplify the logic below, but this `last_scanned_block` diesel type can be null (when used on line 326). So I think that using COALESCE would change the expected behavior when used on line 326. 

WDYT?

Reply:
Were you referring to https://github.com/mobilecoinfoundation/mobilecoin/blob/44072c0e5000da13722ffcefa2ba2b079add13e6/fog/sql_recovery_db/src/lib.rs#L336 by any chance?

If that's what you're talking about then there's no reason why it should be nullable there, https://github.com/mobilecoinfoundation/mobilecoin/blob/44072c0e5000da13722ffcefa2ba2b079add13e6/fog/sql_recovery_db/src/lib.rs#L366 will need to change to `i64` with the understanding that a value of 0 means nothing was scanned (which seems fine to me).

Reply:
Digging deeper into this, it makes its way into `IngressPublicKeyRecord` which does use an `Option<u64>` so perhaps this is not the right approach.

Reply:
SG. As discussed offline, I'll leave the implementation as is.